### DISCUSSION TITLE
v2.7.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#5056] [Releases & Announcements]
## Changelog

- Support for Whatsapp Cloud API
- Ability to create dashboard apps from UI
- New Chatwoot CLI tool
- Support for Twilio Messaging Services
- Support HMAC enforcement in API channel
- Enhanced contact identification and merge logic
- APIs for custom sort
- API for updating contact avatar
- Numerous bug fixes and enhancements

Thanks to @iprithvitharun, @jordan-brough, @gk7700, @SimonHausdorf, @giquieu for the contributions

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v2.7.0'>v2.7.0</a>.</em>

**Comment by SimonVillage** [comment]
Thanks for taking care of the PRs and providing this wonderful open source project.

**Comment by sojan-official** [comment]
## 2.7.0 Dev Changelog

### API Endpoints

#### Additions

- `/api/v1/widget/contact/set_user` [PATCH]
    - PR: https://github.com/chatwoot/chatwoot/pull/4886/files
    - Description: `SetUser` sdk method now uses a dedicated API endpoint, than using the contact update endpoint.
- `/api/v1/accounts/:account_id/contacts/:id` [CREATE, UPDATE]
    - PR: https://github.com/chatwoot/chatwoot/pull/4719
    - description: These endpoints now accespt and `avatar` file object or `avatar_url` value.
- `/api/v1/accounts/:account_id/contacts/:id/avatar` [DELETE]
    - PR: https://github.com/chatwoot/chatwoot/pull/4719
    - description: Ability to delete Contact Avatar
- `webhooks/whatsapp/:phone_number` [POST] 
    - PR: https://github.com/chatwoot/chatwoot/pull/4938
    - Description: Verify endpoint for whatsapp Cloud API
        
#### Deletions
- none

#### Other Changes
  - Knowledge Base Endpoints (Alpha)
    - PR: https://github.com/chatwoot/chatwoot/pull/4955
    - PR: https://github.com/chatwoot/chatwoot/pull/4946
    - PR: https://github.com/chatwoot/chatwoot/pull/4912
    - PR: https://github.com/chatwoot/chatwoot/pull/4879
    - description : These APIs are under development and there could be significant changes

---

### Payload Changes

#### Additions

- Inbox Object 
  - Description: Inbox object will have the following attributes additionally based on appropriate situations:
    `messaging_service_sid: Twilio`,`hmac_token: API` `provider : whatsapp` , `provider_config: whatsapp - admin`
  - PR: https://github.com/chatwoot/chatwoot/pull/4242
  - PR: https://github.com/chatwoot/chatwoot/pull/4807
  - PR: https://github.com/chatwoot/chatwoot/pull/4938
  - PR: https://github.com/chatwoot/chatwoot/pull/4944


#### Deletions

- none
